### PR TITLE
minor bug fixes for gke

### PR DESCRIPTION
**Pull request by ajslone**
This PR implements some minor bugfixes for the gke backend.

### CODE REVIEW

File: caliban/platform/gke/cluster.py [hunk 1]
```diff
@@ -188,13 +188,13 @@ class Cluster(object):
...
   # --------------------------------------------------------------------------
-  def __init__(self, name: str, project_id: str, zone: str,
+  def __init__(self, name: Optional[str], project_id: str, zone: str,
                credentials: Credentials):
     self._cluster_client = None
-    self._gke_cluster = None
-    self._core_api = None
-    self._batch_api = None
-    self._apps_api = None
+    self._gke_cluster: Optional[GKECluster] = None
```
Comment (sritchie) [inline caliban/platform/gke/cluster.py:194]: nice, didn't realize you could type variables inline.

**Comment by sritchie** [comment]
@ajslone also do previous tests cover these? Is there any easy way to lock in this behavior with tests, or do the types cover it?

File: caliban/platform/gke/cluster.py (hunk 1 above)
Comment (ajslone) [reply caliban/platform/gke/cluster.py:194]: now that we require python >= 3.6 we can!

**Comment by ajslone** [comment]
I need to add tests for `gke.cluster` and `gke.cli`, so the changes here for those are not entirely covered/caught by tests yet. The changes for `gke.util` are currently tested in our test suite, but the changes in the PR in that file just modify the code to make `mypy` happy, so in this case our tests make sure that these changes don't break anything.

**Pull request merged**